_data-prepper/pipelines/configuration/processors/mutate-strings.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: default
+title: Mutate strings
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Mutate string processors
+
+Mutate string processesors allow you to change the way that a string appears. For example, you can use the `uppercase_string` processor to convert a string to uppercase, and you can use the `lowercase_string` processor to convert a string to lowercase. The following is a list of processors that allow you to mutate a string:
+
+* [substitute_string](#substitutestringprocessor)
+* [split_string](#splitstringprocessor)
+* [uppercase_string](#uppercasestringprocessor)
+* [lowercase_string](#lowercasestringprocessor)
+* [trim_string](#trimstringprocessor)
+
+## substitute_string
+
+The `substitute_string` processor matches a key's value against a regular expression (regex) and replaces all matches with a replacement string.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - substitute_string:
+        entries:
+          - source: "message"
+            from: ":"
+            to: "-"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` of the file source in your `pipeline.yaml` file with your file path:
+
+```json
+{"message": "ab:cd:ab:cd"}
+```
+
+When you run Data Prepper with the `pipeline.yaml` file, you will see the following output:
+
+```json
+{"message": "ab-cd-ab-cd"}
+```
+
+If the `from` regex string does not return a match, the key returns without any changes.
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
```

Comment:
Are these parameters? How are these items used? What does configuration specifically allow the user to do here?